kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/FailoverUrlList.java

```diff
@@ -24,7 +26,9 @@ public FailoverUrlList(List<String> urls) {
 
   public FailoverUrlList(List<String> urls, int retryGracePeriodInMs) {
     if (urls != null && !urls.isEmpty()) {
-      this.urls = new ArrayList<>(urls);
+      this.urls = urls.stream()
```

Comment:
I would move this logic to upper level to ClusterMapper.setSchemaRegistry method. Also in setSchemaRegistry I would add url validity check to be fail-fast

Reply:
my point is basically in fail-fast check implementation - currently we can't guarantee that this list will actually contain valid urls. If you think that it's better to keep this logic in constructor - no problem.